client/src/common/components/Members/components/MemberDetails.jsx

```diff
@@ -86,20 +87,22 @@ class MemberDetails extends PureComponent {
     }));
   };
 
-  changeCohortRole = selectedRole => {
+  changeCohortRole = () => {
     const {
+      _id: userId,
+      addOwnerRoleInCohort,
+      isOwner,
       match: {
         params: { id }
       },
-      _id: userId,
-      changeRoleInCohort
+      removeOwnerRoleInCohort
     } = this.props;
 
     this.setState({ pending: true });
 
-    changeRoleInCohort(id, userId, selectedRole).finally(() =>
-      this.setState({ pending: false })
-    );
+    const action = isOwner ? removeOwnerRoleInCohort : addOwnerRoleInCohort;
+
```

Comment:
I think we sholud check in those methods whetever current user isOwner and then allow to fire the action
something like this 
```
If(isOwner) { 
//...rest of the code

this.setState(...)
action().....
}
```

here in this method, and down below in `changeListRole` method also.

Reply:
In render method we check if currentUser is an Owner and then disable render options that allow user to change roles and remove other members if he isn't an owner.
In these methods we would use the same props for this checking as we used in render method so I think It's not necessary.